daemon/src/main/java/org/mvndaemon/mvnd/daemon/Server.java

```diff
@@ -655,8 +656,13 @@ public void fail(Throwable t) throws Exception {
 
         @Override
         protected void onStartSession(MavenSession session) {
-            queue.add(new BuildStarted(getCurrentProject(session).getName(), session.getProjects().size(),
-                    session.getRequest().getDegreeOfConcurrency()));
+
+            final DependencyGraph smartGraph = DependencyGraph.fromMaven(session);
+            session.setProjectDependencyGraph(smartGraph);
```

Comment:
@gnodet this is quite a rough way to do it. I was not able to persuade the guice container to prefer my org.apache.maven.graph.GraphBuilder impl. to the default one. So I gave up and did this replacement instead. I would not mind if you veto this and propose something more elegant.